targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_KW41Z/trng_api.c

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Hardware entropy collector for the K64F, using Freescale's RNGA
+ *
+ *  Copyright (C) 2006-2015, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * Reference: "K64 Sub-Family Reference Manual, Rev. 2", chapter 34
+ */
+
+#if defined(DEVICE_TRNG)
+
+#include "fsl_trng.h"
+#include "trng_api.h"
+
+void trng_init(trng_t *obj)
+{
+    trng_config_t trngConfig;
+
+    TRNG_GetDefaultConfig(&trngConfig);
+    /* Set sample mode of the TRNG ring oscillator to Von Neumann, for better random data.*/
+    trngConfig.sampleMode = kTRNG_SampleModeVonNeumann;
+    /* Initialize TRNG */
+    TRNG_Init(TRNG0, &trngConfig);
```

Comment:
According to the manual this function can return with an error. Because of the API we can't propagate this error, but we still need to make it sure that if an error has happened here then any subsequent calls to `trng_get_bytes()` will fail with an error.